hack/update-vendor.sh

```diff
@@ -358,6 +358,13 @@ done
 echo "=== tidying root" >> "${LOG_FILE}"
 go mod tidy >>"${LOG_FILE}" 2>&1
 
+# prune unused pinned non-local replace directives
```

Comment:
this is the same check as on line 338, with the addition of explicitly keeping local replace directives in the root (`k8s.io/foo => ./staging/src/k8s.io/foo`)

Reply:
got it (`select(.New.Path | startswith("./")`)